jenkins/version-def.sh

```diff
@@ -48,9 +60,27 @@ SPARK_REPO=${SPARK_REPO:-"$URM_URL"}
 echo "CUDF_VER: $CUDF_VER, CUDA_CLASSIFIER: $CUDA_CLASSIFIER, PROJECT_VER: $PROJECT_VER \
     SPARK_VER: $SPARK_VER, SCALA_BINARY_VER: $SCALA_BINARY_VER"
 
+# Spark shim versions
+. $(dirname "$0")/common.sh
+case $PHASE_TYPE in
+    pre-release)
+        PROFILE_OPT="-Psnapshots" 
+        ;;
 
-SPARK_SHIM_VERSIONS_STR=${SPARK_SHIM_VERSIONS_STR:-"311 321cdh 312 313 314 320 321 322 330 331"}
-
-IFS=" " <<< $SPARK_SHIM_VERSIONS_STR read -r -a SPARK_SHIM_VERSIONS
-
+    *)
+        PROFILE_OPT="-PnoSnapshots" 
+        ;;
+esac
+# base version
+get_spark_shim_versions $PROFILE_OPT
 SPARK_BASE_SHIM_VERSION=${SPARK_SHIM_VERSIONS[0]}
+# snapshots + noSnapshots
+get_spark_shim_versions -Psnapshots
+SPARK_SHIM_VERSIONS_ALL=("${SPARK_SHIM_VERSIONS[@]}")
+# noSnapshots only
+get_spark_shim_versions -PnoSnapshots
+SPARK_SHIM_VERSIONS_NOSNAPSHOTS=("${SPARK_SHIM_VERSIONS[@]}")
+# build and run unit tests on one 3.1.X version (base version covers this), one 3.2.X and one 3.3.X version
+SPARK_SHIM_VERSIONS_TEST=($(for version in ${SPARK_SHIM_VERSIONS_ALL[@]}; do [[ $version != ${SPARK_BASE_SHIM_VERSION: 0: 2}* ]] && echo $version; done | sort -n -k1.1,1.2 -u -s))
```

Comment:
this part would vary a lot, the code may not work correctly if list shims updated.
I would prefer to simply add a new property like https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/dist/pom.xml#L43-L59 here, so pom should be the only place to modify list of specific shims

Reply:
I am fine with a dedicated property but I think it will work as is. 

somewhat simpler version of this line would be 
```bash
SPARK_FEATURE_VERSIONS=($(<<< "${SPARK_SHIM_VERSIONS[@]}" xargs -n 1 | sort -n -k1.1,1.2 -u -s))
SPARK_SHIM_VERSIONS_TEST=("${SPARK_FEATURE_VERSIONS[@]:1}") 
```